docs/graph-algorithms/prims-algorithm.mdx

```diff
@@ -0,0 +1,255 @@
+}
+/* in main we give inputs to graph array and calls operation function */
+int main() {
+  int all_edges;
+  int array[number];
+  /*memset Sets the first num bytes of the block of memory 
+  pointed by ptr to the specified value faster than loop*/
+  memset(array, false, sizeof(array));
+  all_edges = 0;
+  array[0] = true;
+  cout << "enter the elements into graph array " << endl;
+  /* taking the inputs for graph in form of loop
+  which is of having number(of rows and coloumns) */
+  int row, coloumn;
+  int graph[number][number];
+  for (int i = 0; i < number; i++) {
+    for (int j = 0; j < number; j++) {
+      cin >> graph[i][j];
+    }
+  }
+  cout << "Weight of ele," << " " << "Edge of graph" << "\n";
+  /* iterate while loop till all edges are lesser than num-1 */
+  while (number - 1 > all_edges) {
+    row = 0, coloumn = 0;
+    operation(array, row, coloumn, graph);
+    all_edges++;
+  }
+  return 0;
+}
+```
+
+</TabItem>
+<TabItem value="java">
+
+```java
+// Java Program for Prim's Algorithm
+import java.util.Arrays;
+
+class PGraph {
+
+  public void Prim(int G[][], int V) {
+
+    int INF = 9999999;
+
+    int no_edge; // number of edge
+
+    // create a array to track selected vertex
+    // selected will become true otherwise false
+    boolean[] selected = new boolean[V];
+
+    // set selected false initially
+    Arrays.fill(selected, false);
+
+    // set number of edge to 0
+    no_edge = 0;
+
+    // the number of egde in minimum spanning tree will be
+    // always less than (V -1), where V is number of vertices in
+    // graph
+
+    // choose 0th vertex and make it true
+    selected[0] = true;
+
+    // print for edge and weight
+    System.out.println("Edge : Weight");
+
+    while (no_edge < V - 1) {
+      // For every vertex in the set S, find the all adjacent vertices
+      // , calculate the distance from the vertex selected at step 1.
+      // if the vertex is already in the set S, discard it otherwise
+      // choose another vertex nearest to selected vertex at step 1.
+
+      int min = INF;
+      int x = 0; // row number
+      int y = 0; // col number
+
+      for (int i = 0; i < V; i++) {
+        if (selected[i] == true) {
+          for (int j = 0; j < V; j++) {
+            // not in selected and there is an edge
+            if (!selected[j] && G[i][j] != 0) {
+              if (min > G[i][j]) {
+                min = G[i][j];
+                x = i;
+                y = j;
+              }
+            }
+          }
+        }
+      }
+      System.out.println(x + " - " + y + " :  " + G[x][y]);
+      selected[y] = true;
+      no_edge++;
+    }
+  }
+
+  public static void main(String[] args) {
+    PGraph g = new PGraph();
+
+    // number of vertices in grapj
+    int V = 5;
+
+    // create a 2d array of size 5x5
+    // for adjacency matrix to represent graph
+    int[][] G = { { 0, 9, 75, 0, 0 }, { 9, 0, 95, 19, 42 }, { 75, 95, 0, 51, 66 }, { 0, 19, 51, 0, 31 },
```

Comment:
We need to have dynamic input. Why is it here?

Reply:
@HarshCasper I have taken it from Neoalgo